test/unit/org/apache/cassandra/cql3/validation/operations/CQLVectorTest.java

```
@@ -0,0 +1,142 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.validation.operations;
+
+import org.junit.Test;
+
+import org.apache.cassandra.cql3.CQLTester;
+
+public class CQLVectorTest extends CQLTester.InMemory
+{
+    @Test
+    public void select()
+    {
+        createTable(KEYSPACE, "CREATE TABLE %s (pk vector<int, 2> primary key)");
+
+        execute("INSERT INTO %s (pk) VALUES ([1, 2])");
+
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, 2]"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = ?", vector(1, 2)), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, 1 + 1]"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, ?]", 2), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, (int) ?]", 2), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, 1 + (int) ?]", 1), row(list(1, 2)));
+
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, 2])"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, 2], [1, 2])"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN (?)", vector(1, 2)), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, 1 + 1])"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, ?])", 2), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, (int) ?])", 2), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, 1 + (int) ?])", 1), row(list(1, 2)));
+
+        assertRows(execute("SELECT * FROM %s WHERE pk > [0, 0] AND pk < [1, 3] ALLOW FILTERING"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE token(pk) = token([1, 2])"), row(list(1, 2)));
+    }
+
+    @Test
+    public void insert()
```

a new question : should vectortype support index ？ if we are not going to support ,should we throw an exception ,otherwise a new test case is need ?

---

> should vectortype support index 

in dev@ I argued we should, but know @jbellis was clear he didn't want to, so for this patch I didn't add support.

`vector[?]` is a `Operator`, and since the current operator doesn't know about this type it will fail, so don't think I need a test... I do feel we could add support super trivially so don't like a test blocking this feature for the future...

🤷

---

OK